core/matchdata/deconstruct_keys_spec.rb

```diff
@@ -0,0 +1,63 @@
+require_relative '../../spec_helper'
+
+describe "MatchData#deconstruct_keys" do
+  ruby_version_is "3.2" do
+    it "returns whole hash for nil as an argument" do
+      m = /(?<f>foo)(?<b>bar)/.match("foobar")
+
+      m.deconstruct_keys(nil).should == { f: "foo", b: "bar" }
+    end
+
+    it "returns only specified keys" do
+      m = /(?<f>foo)(?<b>bar)/.match("foobar")
+
+      m.deconstruct_keys([:f]).should == { f: "foo" }
+    end
+
+    it "requires one argument" do
+      -> {
+        /l/.match("l").deconstruct_keys
+      }.should raise_error(ArgumentError)
```

Comment:
**suggestion:** I would move out `/l/.match("l")` from the block. It makes the test more focused on the `.deconstruct_keys` call.

Also I would add error message.